packages/material-ui/src/Grid/Grid.js

```diff
@@ -390,24 +423,23 @@ Grid.propTypes = {
   zeroMinWidth: PropTypes.bool,
 };
 
-const StyledGrid = withStyles(styles, { name: 'MuiGrid' })(Grid);
-
-if (process.env.NODE_ENV !== 'production') {
-  const requireProp = requirePropFactory('Grid');
-  StyledGrid.propTypes = {
-    ...StyledGrid.propTypes,
-    alignContent: requireProp('container'),
-    alignItems: requireProp('container'),
-    direction: requireProp('container'),
-    justifyContent: requireProp('container'),
-    lg: requireProp('item'),
-    md: requireProp('item'),
-    sm: requireProp('item'),
-    spacing: requireProp('container'),
-    wrap: requireProp('container'),
-    xs: requireProp('item'),
-    zeroMinWidth: requireProp('item'),
-  };
-}
+// TODO: This is breaking the buildAPI generation :(
```

Comment:
This is breaking the generation of the props in `buildAPI`. Not sure why it was not an issue before tough.. :\

Reply:
Is there a reason why they are added in the initial definition of the propTypes?  Couldn't we just use `/* @typescript-to-proptypes-ignore */` and add these as part of the definition of the `Grid.propTypes`?

Reply:
@mbrookes any opinion on this?

Reply:
I have tried something but it doesn't fly. I think that it was working before because we had two different components. It's not longer the case. Your proposed solution sounds promising.

Reply:
Ok will try this then tomorrow 👍

Reply:
Refactoring to `chainPropTypes` work, but we will have to manually update the propTypes if we have changes in the props interface. @oliviertassinari I saw you fixed the issue with https://github.com/mui-org/material-ui/pull/24395/commits/edd89cab6e2d66e41291d56e9afe6d90acc245a1 so I'd keep your solution for now, and we can come back in the future if this comes up as an issue again. What do you think?

Reply:
> but we will have to manually update the propTypes if we have changes in the props interface.

The problem with the solution I have tried is that it override the exisiting prop-types. How about we update the approach to extend them instead? For instance, spacing used to be an enum, now it's only about weither the container prop is applied too or note.

Reply:
Ah I see, ok, let me update the replacement to actually use the `chainPropTypes`, considering the default values for the `PropTypes`

Reply:
To remove any potential doubt, my latest proposal was to improve `requirePropFactory(` to do the merge internally.

Reply:
In order to do this inside `requirePropFactory()` we will need to change the API of it, to accept the component where the propTypes are already defined, plus for each prop we will need to provide the name of the prop + the prop it depends on. Are we ok with making this changes? I see that the util is exported so it would be a breaking chage. Luckily on our side the Grid is the only place it is used so it should be safe...

Reply:
Actually the change doesn't have to be breaking, nevermind will update it :)

Reply:
Implemented here https://github.com/mnajdova/material-ui/pull/23 let me know if it's what you had in mind and will merge it as part of this PR